src/hooks/useFavouriteMessages.ts

```diff
@@ -15,27 +14,65 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+import { MatrixEvent } from "matrix-js-sdk/src/models/event";
 import { useState } from "react";
 
-const favouriteMessageIds = JSON.parse(
-    localStorage?.getItem("io_element_favouriteMessages")?? "[]") as string[];
+interface IButtonProp {
+    mxEvent?: MatrixEvent;
+}
+
+let sortState = false;
+let isSearchClicked = false;
+
+export default function useFavouriteMessages(props?: IButtonProp) {
+    const favouriteMessagesIds = JSON.parse(
+        localStorage?.getItem("io_element_favouriteMessages")?? "[]") as any[];
 
-export default function useFavouriteMessages() {
     const [, setX] = useState<string[]>();
+    const eventId = props?.mxEvent.getId();
+    const roomId = props?.mxEvent.getRoomId();
+    const content = props?.mxEvent.getContent();
 
     //checks if an id already exist
-    const isFavourite = (eventId: string): boolean => favouriteMessageIds.includes(eventId);
+    const isFavourite = (): boolean => {
+        return favouriteMessagesIds.some(val => val.eventId === eventId);
+    };
 
-    const toggleFavourite = (eventId: string) => {
-        isFavourite(eventId) ? favouriteMessageIds.splice(favouriteMessageIds.indexOf(eventId), 1)
-            : favouriteMessageIds.push(eventId);
+    const toggleFavourite = () => {
+        isFavourite() ? favouriteMessagesIds.splice(favouriteMessagesIds.findIndex(val => val.eventId === eventId), 1)
+            : favouriteMessagesIds.push({ eventId, roomId, content });
 
         //update the local storage
-        localStorage.setItem('io_element_favouriteMessages', JSON.stringify(favouriteMessageIds));
+        localStorage.setItem('io_element_favouriteMessages', JSON.stringify(favouriteMessagesIds));
 
         // This forces a re-render to account for changes in appearance in real-time when the favourite button is toggled
         setX([]);
     };
 
-    return { isFavourite, toggleFavourite };
+    const reorderFavouriteMessages = () => {
+        sortState = !sortState;
+    };
+
+    const setSearchState = (val: boolean) => {
+        isSearchClicked = val;
+    };
+
+    const clearFavouriteMessages = () => {
+        localStorage.removeItem('io_element_favouriteMessages');
+    };
+
+    const getFavouriteMessagesIds = () => {
+        return sortState ? favouriteMessagesIds.reverse(): favouriteMessagesIds;
+    };
+
+    return {
+        isFavourite,
+        toggleFavourite,
+        getFavouriteMessagesIds,
+        reorderFavouriteMessages,
+        clearFavouriteMessages,
+        setSearchState,
+        isSearchClicked,
+
+    };
```

Comment:
With all these things, it feels like these should be a store rather than a hook perhaps :thinking: When is the switch to account data going to happen?

Reply:
Yeah, i figured out i needed a store instead of this, it's really a pain doing this with hooks only.

Reply:
Regarding when is the switch to AccountData, i'll start working on it starting next month. For now i think we should stick to this.